### PR TITLE
chore(ci): check branch name for '@' and '#'

### PR DESCRIPTION
**Pull request by mroz22**

### CODE REVIEW

File: ci/environment.yml [hunk 1]
```diff
@@ -19,4 +19,5 @@ build & push image:
 install:
   stage: setup environment
   script:
+    - ci/scripts/check_branch_name.sh
```
Comment (prusnak) [inline ci/environment.yml]: Shouldn't we pass `"$CI_COMMIT_BRANCH"` as first argument to the script here?

File: ci/scripts/check_branch_name.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+
+name=$(git rev-parse --abbrev-ref HEAD)
```
Comment (prusnak) [inline ci/scripts/check_branch_name.sh]: We don't need to do this if we pass `"$CI_COMMIT_BRANCH"` as argument.

File: ci/scripts/check_branch_name.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+STR='GNU/Linux is an operating system'
+SUB='Linux'
+
+echo "branch name:"
+echo $name
```
Comment (prusnak) [inline ci/scripts/check_branch_name.sh]: We can drop these 5 lines, no?

File: ci/environment.yml [hunk 4]
```diff
@@ -19,4 +19,5 @@ build & push image:
 install:
   stage: setup environment
   script:
+    - ci/scripts/check_branch_name.sh ${CI_COMMIT_BRANCH}"
```
Comment (prusnak) [inline ci/environment.yml]: Typo:
```suggestion
    - ci/scripts/check_branch_name.sh ${CI_COMMIT_BRANCH}
```

File: ci/scripts/check_branch_name.sh [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+echo "branch name:"
+echo $1
+
+if [[ "$1" == *"@"* || "$name" == *"#"* ]]; then
```
Comment (prusnak) [inline ci/scripts/check_branch_name.sh]:
```suggestion
if [[ "$1" == *"@"* || "$1" == *"#"* ]]; then
```

File: ci/scripts/check_branch_name.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+echo "branch name:"
+echo $1
```
Comment (prusnak) [inline ci/scripts/check_branch_name.sh]:
```suggestion
echo "$1"
```

**Review by prusnak — DISMISSED** [review]

**Review by matejkriz — APPROVED** [review]
let's squash and merge

**Review by vdovhanych — COMMENTED** [review]
Looks ok but wouldn't it be better to have it in git hooks also?

**Comment by mroz22** [comment]
> Looks ok but wouldn't it be better to have it in git hooks also?

I think we are only bother by these symbols in gitlab jobs (deploys, tests), it shouldn't be breaking anything in github jobs.

**Comment by matejkriz** [comment]
> > Looks ok but wouldn't it be better to have it in git hooks also?
> 
> I think we are only bother by these symbols in gitlab jobs (deploys, tests), it shouldn't be breaking anything in github jobs.

I think @vdovhanych meant git commit hook (precommit/prepush) and it could make sense but I believe this gitlab check is really sufficient for us especially because the fix is very easy (just renaming the branch). So the whole point is to get some reasonable message what is wrong and why pipelines are failing...

**Review by vdovhanych — APPROVED** [review]

**Pull request merged**